dora/src/bytecode/generator.rs

```diff
@@ -1548,6 +1549,13 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
 
                 Intrinsic::GenericArrayNew => self.emit_intrinsic_new_array(expr, dest),
 
+                Intrinsic::GenericArrayWithValues => {
+                    let ty = self.ty(expr.id);
+                    let array_reg = self.emit_array_with_variadic_arguments(expr, &[ty], 0, dest);
+                    self.registers.free_if_temp(array_reg);
```

Comment:
I don't think this line is correct.

Reply:
So this line needs to happen after the caller got the register (and likely did something with it), right?

Reply:
I think you can simply remove that line. array_reg can't be freed at that point, it should be freed by the caller later. Unfortunately there is no check in the bytecode generator yet that makes sure that instruction operands are still live and not freed.

Reply:
Btw the caller should already free the register, so you shouldn't need to change any other code.

Reply:
I think without that line I ran into the assert – I'll check again.

Reply:
Weeeeird – it works! I would have sworn that didn't work earlier ...

Reply:
That was without the dest argument, right?